test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/Parallel/ParallelRunDataAggregatorTests.cs

```diff
@@ -119,34 +119,61 @@ public void AggregateShouldAggregateInvokedCollectorsCorrectly()
         Assert.AreEqual(invokedDataCollectors2[0].Uri, aggregator.InvokedDataCollectors[1].Uri);
     }
 
+    [TestMethod]
+    public void AggregateShouldAggregateTelemetryEventsCorrectly()
+    {
+        var aggregator = new ParallelRunDataAggregator(Constants.EmptyRunSettings);
+
+        var telemetryEvents = new Collection<TelemetryEvent>()
+        {
+            new TelemetryEvent("event", new Dictionary<string, object>())
+        };
+
+        aggregator.Aggregate(null, null, null, TimeSpan.Zero, false, false, null, null, null, telemetryEvents, null);
+        Assert.AreEqual(1, aggregator.TelemetryEvents.Count, "TelemetryEvents List must have data.");
+
+        aggregator.Aggregate(null, null, null, TimeSpan.Zero, false, false, null, null, null, telemetryEvents, null);
+        Assert.AreEqual(2, aggregator.TelemetryEvents.Count, "TelemetryEvents List must have aggregated data.");
+
+        aggregator.Aggregate(null, null, null, TimeSpan.Zero, false, false, null, null, null, telemetryEvents, null);
+        Assert.AreEqual(3, aggregator.TelemetryEvents.Count, "TelemetryEvents List must have aggregated data.");
+
+        Assert.AreEqual(telemetryEvents[0].Name, aggregator.TelemetryEvents[0].Name);
+        Assert.AreEqual(telemetryEvents[0].Name, aggregator.TelemetryEvents[1].Name);
```

Comment:
Shouldn't the aggregator overwrite the TelemetryEvents[0] for each of the subsequent calls ?

Reply:
no, each telemetry event is new event - we do not merge it